pkg/controller/common/util.go

```diff
@@ -1238,6 +1241,23 @@ func NewImportDataVolume(name string) *cdiv1.DataVolume {
 	}
 }
 
+// GetDataVolumeStamp encodes a DataVolume manifest as string
+func GetDataVolumeStamp(dv *cdiv1.DataVolume) (string, error) {
+	dvStamp := &cdiv1.DataVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            dv.Name,
+			Namespace:       dv.Namespace,
+			OwnerReferences: dv.OwnerReferences,
```

Comment:
Realistically, this scheme only "works" for DataVolumeTemplates in VMs.  Not so much for stand alone DVs

Reply:
Why? we can use the complete DV for the scheme (excluding `UID, ResourceVersion, Generation, CreationTimestamp, ManagedFields, Status`), or add any fields we want.

Reply:
Once a DV is garbage collected, there will never be another DV with with the same uid/resourceversion/generation/creationtimestamp.  So you will not be getting the idempotency guarantees you want

Reply:
I agree about the first part, but you will get another DV with the same name/ns/spec which we can mutate annotating it with `deleteAfterCompletion = false` to prevent a create/delete loop, and the PVC can be updated to have the DV ownerRefrence.

Reply:
I am also not sure why this fails with stand alone DVs, probably missing something

Reply:
With stand alone DVs which are managed by argo will become out of sync when the DV is garbage collected. Which then makes argo create a new one, which currently fails because the PVC already exists. Secondly if the application is removed from the cluster by accident, deleting the DV, will cause the PVC to also get deleted by CDI. This problem doesn't exist in the DVTemplate case because the DV is not managed by argo, and thus never gets out of sync.